client/components/data/query-active-promotions/index.jsx

```diff
@@ -1,45 +1,25 @@
-/** @format */
-
 /**
  * External dependencies
  */
-
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { isRequestingActivePromotions } from 'state/active-promotions/selectors';
 import { requestActivePromotions } from 'state/active-promotions/actions';
 
-class QueryActivePromotions extends Component {
-	UNSAFE_componentWillMount() {
-		if ( ! this.props.requestingActivePromotions ) {
-			this.props.requestActivePromotions();
+export default function QueryActivePromotions() {
+	const requestingActivePromotions = useRef( useSelector( isRequestingActivePromotions ) );
+	const dispatch = useDispatch();
+
+	// Only runs on mount.
+	useEffect( () => {
```

Comment:
Do you agree with moving dispatches after browser rendering, by using `useEffect` instead of `useLayoutEffect`/`componentDidMount`/`componentDidUpdate`?

Reply:
Totally yes. Issuing the requests can wait after the UI is rendered and the user sees the up-to-date UI with placeholders etc.